### DISCUSSION TITLE
Unparse method

### DISCUSSION THREAD

**Discussion by yudinmaxim** [open-circle/valibot#399] [Q&A]
Hi everyone.

I want to do the following:

Get data from the API - perform data parsing using safeParce(schema, data) - validate and transform the data - get data with camelCase names. (This is already implemented and works well and I like it)
Create/modify form data received from the API and send the values back.
I understand how to convert data: api - valibot parse - data, but I don't understand how to do the reverse conversion: data - valibot - API.

So, I want the schema to be able to transform data not only from API - data, but also in the opposite direction.

For example, if I perform a field name transformation during data validation: field_name -> fieldName, then when performing data transfer to the API, I need to make the reverse transformation: fieldName -> field_name.

Does valibot have the ability to perform such transformations?

```javascript
export const presetSchema = object({
  id: string([
    minLength(1),
  ]),
  parentId: optional(string()), // in API this field name is 'parent_id'
  params: object({
    audio: optional(number()),
    prefix: optional(string()),
    volte: optional(boolean())
  })
})
export type tPreset = Output<typeof presetSchema>

export const presetsSchema = coerce(
  array(presetSchema),
  (input: any): tPreset => {
    const output = {
      ...input,
      parentId: input.parent_id
    }
    delete output.parentId
    return output
  }
)```

**Comment by fabian-hiller** [comment]
Thank you for creating this discussion. Unfortunately, this is not supported at this time. Also, I have no idea how we could support this with a nice DX. Feel free to make suggestions. As a workaround, I would create two schemas. One for each direction.

**Reply by yudinmaxim** [reply]
Thank you for responce.
My solution is as follows.

https://stackblitz.com/edit/typescript-5yjee6?file=index.ts

Yes, two schemes are used, which looks reasonable:
1. Scheme for validating data received from the API
2. Scheme for validating data sent to the API

I'm not sure about the `pick` solution - it's confusing that you have to manually specify a list of required fields - but I haven't found any other solution.
If there are any suggestions, I would be very happy.

**Reply by fabian-hiller** [reply]
I think you should replace `coerce` with `transform`. You can read more about both methods [here](https://valibot.dev/guides/methods/). Also the use of `pick` and `merge` seams to be unnecessary in this use case. `pick` is similar to `Pick` in TypeScript.

**Reply by yudinmaxim** [reply]
Thanks for the feedback.
I realized that I should rethink this decision and the logic of working with validators in general a few more times))
Thank you very much for your help in this matter.
